[Spaces] Show callout on spaces permission tab when security is disabled

kowalczyk-krzysztof opened the pull request:
## Summary

This PR updates the behavior of `Permissions` tab in `Space Management` when `xpack.security.enabled` is set to `false` to show a callout with a meaningful explanation.
![Screenshot 2025-02-17 at 12 42 58](https://github.com/user-attachments/assets/881d0bbe-e30b-41b4-8c0a-5b8e127786ab)


Closes: #210241

kowalczyk-krzysztof committed:
Show callout on spaces permission tab when security is disabled

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

kowalczyk-krzysztof committed:
Merge branch 'main' into fix/disable-permission-tab-if-secruity-is-disabled

elasticmachine commented:
Pinging @elastic/kibana-security (Team:Security)

florent-leborgne commented on x-pack/platform/plugins/shared/spaces/public/management/edit_space/security_disabled_callout.tsx:
```diff
@@ -0,0 +1,30 @@
+    {i18n.translate(
+      'xpack.spaces.management.spaceDetails.contentTabs.securityDisabledDescription',
+      {
+        defaultMessage: 'To manage permissions, enable security feature.',
```
```suggestion
        defaultMessage: 'To manage space permissions, you must enable security features first.',
```

Wondering if it would be useful to link to https://www.elastic.co/guide/en/elasticsearch/reference/current/security-minimal-setup.html#_enable_elasticsearch_security_features in addition?

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/276736)
* Commit: a60f8193ddf5c86e2f4b555d33055411c1258f5a

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/0ecbbd52becb2d48150c837149d54125ddd5c6e3) | [after](https://github.com/elastic/kibana/commit/a60f8193ddf5c86e2f4b555d33055411c1258f5a) | diff |
| --- | --- | --- | --- |
| `spaces` | 359 | 360 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/0ecbbd52becb2d48150c837149d54125ddd5c6e3) | [after](https://github.com/elastic/kibana/commit/a60f8193ddf5c86e2f4b555d33055411c1258f5a) | diff |
| --- | --- | --- | --- |
| `spaces` | 269.7KB | 270.2KB | +517.0B |



cc @kowalczyk-krzysztof

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276736","number":276736,"commit":"a60f8193ddf5c86e2f4b555d33055411c1258f5a"}],"number":276736}
buildkite-pr-comment-->

florent-leborgne reviewed (commented):
It sounds a bit better to me when in plural like in [the docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-minimal-setup.html#_enable_elasticsearch_security_features)

Would it be possible to add a link [using the doc links service](https://github.com/elastic/kibana/blob/24b75740fd08f0a3c3ba36f3fa67782e1c2b7c1a/src/platform/packages/shared/kbn-doc-links/src/get_doc_links.ts#L666) to that page to add some (maybe) useful content available in addition?

kowalczyk-krzysztof committed:
Add docs link

kowalczyk-krzysztof committed:
Update test

kowalczyk-krzysztof commented:
@florent-leborgne 
I added your suggestions and added the relevant documentation to docs links. For now I simply put the `xpack.spaces.management.spaceDetails.contentTabs.securityDisabledLinkText` as the text for the documentation link.

![Screenshot 2025-02-17 at 12 04 37](https://github.com/user-attachments/assets/ca3cf009-6ae0-4e51-b034-39ebe3cc1dd0)

florent-leborgne commented:
@kowalczyk-krzysztof Thanks! Usually we'd put the link as a "Learn more" like this: 

`To manage space permissions, you must enable security features first. {Learn more}`

kowalczyk-krzysztof commented:
> @kowalczyk-krzysztof Thanks! Usually we'd put the link as a "Learn more" like this:
> 
> `To manage space permissions, you must enable security features first. {Learn more}`

Right, that makes sense. I'll edit that.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/276832)
* Commit: d18a0ff3cf049da7403c3995efd3cded821b3b94

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/276832#0195139a-97b4-4990-a2e2-7852c338b724)

### History
* :green_heart: [Build #276736](https://buildkite.com/elastic/kibana-pull-request/builds/276736) succeeded a60f8193ddf5c86e2f4b555d33055411c1258f5a

cc @kowalczyk-krzysztof

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276832","number":276832,"commit":"d18a0ff3cf049da7403c3995efd3cded821b3b94"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276736","number":276736,"commit":"a60f8193ddf5c86e2f4b555d33055411c1258f5a"}],"number":276832}
buildkite-pr-comment-->

kowalczyk-krzysztof committed:
Change to learn more

florent-leborgne commented on x-pack/platform/plugins/shared/spaces/public/management/edit_space/security_disabled_callout.tsx:
```diff
@@ -0,0 +1,43 @@
+          <EuiLink href={enableSecurityLink} target="_blank" external={true}>
+            <FormattedMessage
+              id="xpack.spaces.management.spaceDetails.contentTabs.securityDisabledLearnMore"
+              defaultMessage="Learn more."
```
```suggestion
              defaultMessage="Learn more"
```
Very last comment, then LGTM :)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/276839)
* Commit: aeeadadd29aa794f8bdef7f28ad0ce3a8dad3f03

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/276839#019513ab-2163-494a-ba3b-0004d34c4983)

### History
* :broken_heart: [Build #276832](https://buildkite.com/elastic/kibana-pull-request/builds/276832) failed d18a0ff3cf049da7403c3995efd3cded821b3b94
* :green_heart: [Build #276736](https://buildkite.com/elastic/kibana-pull-request/builds/276736) succeeded a60f8193ddf5c86e2f4b555d33055411c1258f5a

cc @kowalczyk-krzysztof

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276839","number":276839,"commit":"aeeadadd29aa794f8bdef7f28ad0ce3a8dad3f03"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276832","number":276832,"commit":"d18a0ff3cf049da7403c3995efd3cded821b3b94"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276736","number":276736,"commit":"a60f8193ddf5c86e2f4b555d33055411c1258f5a"}],"number":276839}
buildkite-pr-comment-->

kowalczyk-krzysztof committed:
Update tests

kowalczyk-krzysztof committed:
Remove puncutation

kowalczyk-krzysztof committed:
Pass only relevant link, not all docLinks

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/276878)
* Commit: b266d03412154ad78eed01080c4dc52847ab5000

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/0ecbbd52becb2d48150c837149d54125ddd5c6e3) | [after](https://github.com/elastic/kibana/commit/b266d03412154ad78eed01080c4dc52847ab5000) | diff |
| --- | --- | --- | --- |
| `spaces` | 359 | 360 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/0ecbbd52becb2d48150c837149d54125ddd5c6e3) | [after](https://github.com/elastic/kibana/commit/b266d03412154ad78eed01080c4dc52847ab5000) | diff |
| --- | --- | --- | --- |
| `aiAssistantManagementSelection` | 92.2KB | 92.3KB | +110.0B |
| `lists` | 136.2KB | 136.3KB | +110.0B |
| `spaces` | 269.7KB | 270.6KB | +899.0B |
| total |  |  | +1.1KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/0ecbbd52becb2d48150c837149d54125ddd5c6e3) | [after](https://github.com/elastic/kibana/commit/b266d03412154ad78eed01080c4dc52847ab5000) | diff |
| --- | --- | --- | --- |
| `core` | 474.1KB | 474.2KB | +110.0B |
| `spaces` | 28.6KB | 28.7KB | +87.0B |
| total |  |  | +197.0B |



### History
* :broken_heart: [Build #276839](https://buildkite.com/elastic/kibana-pull-request/builds/276839) failed aeeadadd29aa794f8bdef7f28ad0ce3a8dad3f03
* :broken_heart: [Build #276832](https://buildkite.com/elastic/kibana-pull-request/builds/276832) failed d18a0ff3cf049da7403c3995efd3cded821b3b94
* :green_heart: [Build #276736](https://buildkite.com/elastic/kibana-pull-request/builds/276736) succeeded a60f8193ddf5c86e2f4b555d33055411c1258f5a

cc @kowalczyk-krzysztof

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276878","number":276878,"commit":"b266d03412154ad78eed01080c4dc52847ab5000"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276839","number":276839,"commit":"aeeadadd29aa794f8bdef7f28ad0ce3a8dad3f03"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276832","number":276832,"commit":"d18a0ff3cf049da7403c3995efd3cded821b3b94"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/276736","number":276736,"commit":"a60f8193ddf5c86e2f4b555d33055411c1258f5a"}],"number":276878}
buildkite-pr-comment-->

elena-shostak reviewed (approved):
LGTM
